src/sql/arrow_sql_gen/statement.rs

```diff
@@ -894,6 +942,63 @@ fn insert_timestamp_into_row_values(
     }
 }
 
+macro_rules! serialize_list_values {
+    ($data_type:expr, $list_array:expr, $array_type:ty, $vec_type:ty) => {{
+        let mut list_values: Vec<$vec_type> = vec![];
+        if let Some(array) = $list_array.as_any().downcast_ref::<$array_type>() {
+            for i in 0..array.len() {
+                list_values.push(array.value(i).into());
+            }
+        }
+
+        serde_json::to_string(&list_values).map_err(|e| {
+            Error::FailedToCreateInsertStatement {
+                source: Box::new(e),
+            }
+        })?
+    }};
+}
+
+
+fn insert_list_into_row_values_json(
+    list_array: Arc<dyn Array>,
+    list_type: &Arc<Field>,
+    row_values: &mut Vec<SimpleExpr>,
+) -> Result<()> {
+    // arrow_json based serialization
```

Comment:
Note: can potentially use `arrow_json::ArrayWriter`, but it produces a non-flat array, which is probably good for some complex types serialization but not for Arrays/Lists IMO

```
"[{"item":1},{"item":2},{"item":3}]"
```